source/extensions/common/wasm/context.cc

```diff
@@ -1654,6 +1654,14 @@ Http::FilterDataStatus Context::decodeData(::Envoy::Buffer::Instance& data, bool
   }
   if (buffering_request_body_) {
     decoder_callbacks_->addDecodedData(data, false);
```

Comment:
I didn't review the previous PR, but I'm genuinely confused why this isn't causing doubling the data somehow.

Reply:
the add* actually use move semantics

Reply:
Probably the  request_body_buffer_ = &data;  should be in the else, it is not used if buffering. and the if should check the callback pointer is not null (probably never happends to be null) but to be consistent with the place  it is used getBuffer:

https://github.com/envoyproxy/envoy/blob/main/source/extensions/common/wasm/context.cc#L818